frontend/interfaces/schedulable_query.ts

```diff
@@ -0,0 +1,110 @@
+import { IFormField } from "./form_field";
+import { IPack } from "./pack";
+
+// Query itself
+export interface IScheduleableQuery {
+  created_at: string;
+  updated_at: string;
+  id: number;
+  name: string;
+  description: string;
+  query: string;
+  team_id: number | null;
+  interval: number;
+  platform: string; // TODO: Specify strings
+  min_osquery_version: string;
+  automations_enabled: boolean;
+  logging: QueryLoggingOption;
+  saved: boolean;
+  author_id: number;
+  author_name: string;
+  author_email: string;
+  observer_can_run: boolean;
+  packs: IPack[];
+}
+
+// API shapes
+
+// Get a query by id
+// GET /api/v1/fleet/queries/{id}`
+export interface IGetQueryResponse {
+  query: IScheduleableQuery;
+}
+
+// List global or team queries
+// GET /api/v1/fleet/queries?order_key={column_from_queries_table}&order_direction={asc|desc}&team_id={team_id}
+export interface IListQueriesResponse {
+  queries: IScheduleableQuery[];
+}
+
+// Create a new query
+// POST /api/v1/fleet/queries
+export interface ICreateQueryRequestBody {
+  name: string;
+  query: string;
+  description?: string;
+  observer_can_run?: boolean;
+  team_id?: number; // global query if ommitted
+  interval?: number; // default 0 means never run
+  platform?: string; // comma-separated platform compatibility, default "all" if ommitted
+  min_osquery_version?: string; // default all versions if ommitted
+  automations_enabled?: boolean; // whether to send data to the configured log destination according to the query's `interval`.
+  logging?: QueryLoggingOption;
+}
+
+// response is IScheduleableQuery // better way to indicate this?
+
+// Modify a query by id
+// PATCH /api/v1/fleet/queries/{id}
+export interface IModifyQueryRequestBody
+  extends Omit<ICreateQueryRequestBody, "name" | "query"> {
+  id: number;
+  name?: string;
+  query?: string;
+}
+
+// response is IScheduleableQuery // better way to indicate this?
+
+// Delete a query by name
+// DELETE /api/v1/fleet/queries/{name}
+export interface IDeleteQueryRequestBody {
+  team_id?: number; // searches for a global query if ommitted
+}
+
+// Delete a query by id
+// DELETE /api/v1/fleet/queries/id/{id}
+// (no body)
+
+// Delete queries by id
+// POST /api/v1/fleet/queries/delete
+export interface IDeleteQueriesRequestBody {
+  ids: number[];
+}
+
+export interface IDeleteQueriesResponse {
+  deleted: number; // number of queries deleted
+}
+
+// @RachelElysia, what is this interface for?
```

Comment:
This is the data posted to the API for creating a new query or updating a query in `onCreateQuery`, `onUpdateQuery`, `onSaveQueryFormSubmit`